### DISCUSSION TITLE
--bootset boot_behaviour=B4

### DISCUSSION THREAD

**Discussion by dshaw619** [gitbls/sdm#45] [Q&A] [answered]
My sdm --customize command includes the switch: 
`--bootset serial=0,boot_behaviour=B4,boot_wait=0`

I do not have any monitor connected and set up RealVNC with --plugin vnc:"realvnc=1280x720" on the subsequent --burn command (usually I use RaspiOS with Desktop and VNC from one of my other computers to connect;  occasionally I install only RaspiOS Lite so put the plugin in the --burn instead of the --customize).

I am able to customize, burn, and boot a uSD card successfully, but after every restart I am presented with a reduced size window and login prompt.  If I go into raspi-config and set B4 Desktop Autologin my screen dimension are reset to what I defined and I don't get the login prompt on subsequent reboots.

Looking through the messages from the --customize step, I found the message:
`>%Desktop Autologin setting 'B3' will be re-enabled after First Boot`

Am I making a simple mistake or is sdm not handling the B4 setting?

**Comment by gitbls** [comment]
There might be a buglet lurking there. Will have a look...

**Comment by gitbls** [comment]
At the moment the settings for boot behavior in `--bootset` are not tightly tied in with the boot behavior management for the display manager and `--autologin`, with the result being odd behaviors like you are seeing.

Could you provide your full sdm --customize command line and --burn command line so that I can work with it to see what can be done to improve this? Of course remove any personal information.

Also, are you using the 1piboot.conf config file at all?

**Reply by dshaw619** [reply]
```
sudo /usr/local/sdm/sdm --customize \
--wpa ~/Documents/Briefcase/It/Raspberry_Pi/Setup/11-Bullseye/wpa_supplicant.conf \
--keymap us \
--locale en_US.UTF-8 \
--timezone America/Los_Angeles \
--wifi-country US \
--disable piwiz \
--svc-disable userconfig \
--bootset serial=0,boot_behaviour=B4,boot_wait=0 \
--lxde-config lxterminal:/home/doug/Documents/Briefcase/It/Raspberry_Pi/Setup/11-Bullseye/lxterminal.conf \
--restart \
--hostname raspberrypi \
--user des \
--extend --xmb 1024 \
~/Downloads/bullseye-armhf-customized.img
```
```
sudo /usr/local/sdm/sdm --burn /dev/sdd \
--ddsw bs=16M \
--hostname KeyLime \
--plugin vnc:"realvnc=1280x720" \
--expand-root \
~/Downloads/bullseye-armhf-customized.img
```
No, I am not using a custom 1piboot.conf file.  I prefer to not have settings scattered about in multiple places which aren't obvious when reviewing how an .img was configured.  e.g. It took me several minutes to find just where 1piboot.conf was (it doesn't seem to be mentioned at https://github.com/gitbls/sdm/wiki/Bootset-and-1piboot).

**Comment by gitbls** [comment]
Thanks. Will see what can be done.

**Comment by gitbls** [comment]
This will be fixed in the next release. Thanks for pointing it out!

As an aside, using `--hostname` on the customize command is a no-op. It is only processed when burning.

**Comment by gitbls** [comment, marked as answer]
This should work correctly in V7.3, posted today.

**Reply by dshaw619** [reply]
Finally getting back to working with sdm and can confirm this is working correctly.  Thanks.

**Comment by gitbls** [comment]
Thanks for confirming!
